app/src/main/java/com/example/twofragmentactivity/MoviesDataRepository.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.twofragmentactivity
+
+import android.content.Context
+import androidx.lifecycle.MutableLiveData
+import com.google.gson.GsonBuilder
+import java.io.File
+import java.io.IOException
+
+
+interface MoviesDataRepository<T> {
+ fun getMoviesList() : MutableLiveData<List<T>>
+}
+
+class MoviesDataRepositoryImpl (val context: Context?): MoviesDataRepository<MoviesListResponse>
```

Comment:
use application context here. Check how to use application context without getting it from activity context.